### ISSUE TITLE
Discussion: Proposition to change the license from GPL v2.0 to Mozilla Public License v2.0

### ISSUE THREAD

**Issue by maxme** [WordPress/gutenberg#23651]
This is a proposal to change the Gutenberg license, from [GNU General Public License, v2 (GPL v2)](https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html) to [Mozilla Public License v2.0 (MPL v2.0)](https://www.mozilla.org/en-US/MPL/2.0/).

## Reasons for switching from GPL v2.0

Gutenberg supports two main platforms, the web and the native mobile apps. The GPL v2.0 license is a blocker for distributing the Gutenberg library in proprietary mobile apps. Currently the only known users of Gutenberg on mobile are the WordPress mobile apps which are under GPL v2.0 ([WordPress for Android](https://github.com/wordpress-mobile/WordPress-Android), [WordPress for iOS](https://github.com/wordpress-mobile/WordPress-iOS)). Mobile apps under the GPL v2.0 are not common and this limits Gutenberg usage in many apps.

Rich text editor libraries in the mobile space are lacking. There is no well known open source rich text editor for Android or iOS. We believe that Gutenberg could be a key library for many mobile apps, but that will never happen with the GPL v2.

The WordPress-Mobile organisation is in the process of switching the license on [AztecEditor-Android](https://github.com/wordpress-mobile/AztecEditor-Android) and [AztecEditor-iOS](https://github.com/wordpress-mobile/AztecEditor-iOS) (currently GPL v2) for the same reason.

## Why the Mozilla Public License v2.0 (MPL v2.0)

- It's less restrictive than the GPL and LGPL. MPL licensed software can be combined with other material if the files are separated (see paragraph 1.7 `Larger Work` and 3.3 `Distribution of a Larger Work` in the MPL v2.0). This allows proprietary mobile apps to integrate Gutenberg.
- It's a copyleft license, like the GPL, every derivative work must be licensed under the same license. Unlike the BSD or MIT licenses, no one can distribute the code under different terms. It forces anyone modifying Gutenberg to contribute back with their changes (bug fix, new feature, etc.).
- It's compatible with the GPL v2 (see paragraph 1.12 `Secondary License` and 3.3. `Distribution of a Larger Work` in the MPL v2.0), so integration into WordPress and the WordPress apps won't be a problem.

The purpose of this ticket is to get Gutenberg contributors' thoughts on the license change.

**Comment by gziolo** [comment]
There is nearly 80 WordPress packages at the moment. It feels like all those that have a note in README file thar are WordPress specific should stay on “GPL 2.0 or later”, e.g. `@wordpress/editor`,`@wordpress/edit-post`, `@wordpress/edit-site` or `@wordpress/edit-widgets`, etc. There is also a set of packages used only for development purposes that could be relicensed as MIT, e.g. `@wordpress/jest-puppeteer-axe`. My point is that we could continue using GPL as a default license but be more liberal in choosing different license types for smaller subset of code. It would be great to take all that into account as well.

**Comment by radex** [comment]
Hi! I'm so incredibly happy to see this discussion! (I previously proposed doing this [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1427)).

I'm going to throw in my 2 cents into the hat, in case this makes any difference.

For context: I'm the tech lead for [Nozbe Teams](https://nozbe.com), a collaborative todo app. Our tech stack is essentially React on web and React Native on iOS and Android. We're a tiny company, and so we share >80% of app's codebase between these 3 platforms.

Our app sorely lacks a WYSIWYG editor. We had a working implementation on web, but we decided to scrap it, because there was no way to port it on iOS and Android. There are pretty much no viable rich text editors for iOS or Android, yet alone both. But even then, shipping three completely separate, but somehow compatible editors would be a vast amount of work.

And you know yourself, making a good rich text editor from scratch is incredibly hard. It's not something we could do ourselves. 

Gutenberg (or even just parts of it like AztecEditor) could solve this problem for us, if it wasn't for the licensing.

It's 100% understandable why you don't want companies to profit off of your work by releasing it under MIT. I support that decision.

Of course I am biased, but MPL (or LGPL) solves most of the downside for you, while exposing you to the upside.

While we might not be big enough to be able to tackle a challenge of developing a rich text editor from scratch, we're big enough to contribute features and bug fixes to open source projects (and we do - for react-native, multiple RN community packages, and our own two popular frameworks for React apps: [WatermelonDB](https://github.com/Nozbe/WatermelonDB) and [zacs](https://github.com/Nozbe/zacs)).

I'm sure there are many organizations like this. As was noted in the top comment, no one outside of WordPress uses Gutenberg, while also there is no viable alternative other than Gutenberg. Multi-platform development keeps growing in strength, so there is a lot of thirst for a great multi-platform rich text editor. If it was an option for companies to use, I think a lot would. And yeah, they'd get something for free, but they'd be forced to contribute - to you- any improvements they make.

**Comment by maxme** [comment]
@gziolo good point, it makes sense to keep some packages under the GPL. Ideally we would have a way to bundle a GPL-free version of gutenberg (for the web and mobile).

**Comment by hsource** [comment]
I realize this has succeeded already for the Android Aztec Editor! That's great!

@maxme - is there any chance that the [`react-native-aztec`](https://github.com/WordPress/gutenberg/tree/master/packages/react-native-aztec) editor in this repository might have its license changed as well? We were thinking of writing our own adapter to take advantage of the new MPL-licensed Android package, but it'd save us work if we could reuse and contribute to the existing package.

**Comment by maxme** [comment]
@hsource yes we can do that after the license on the iOS version of the Aztec Editor is changed.

**Comment by maxme** [comment]
We got other inputs and dual licensing Gutenberg under GPL v2.0 + MPL v2.0, even if it's a bit redundant, it seems like the best solution for WordPress users and contributors.

It's redundant because the MPL v2.0 is explicit about compatibility with GPL v2.0. Dual licensing will make this compatibility crystal clear for everyone.

I updated the title and the content of that ticket.

**Comment by hsource** [comment]
The iOS license just got merged! I'm pushing for my company to use React Native Aztec instead of rolling our own rich text editor, and a change to the react-native-aztec license would help with that! What needs to happen for that?

**Comment by KestutisIT** [comment]
Finally some great news. Actually that is one of the reasons why in [SolidMVC micro-framework](https://github.com/SolidMVC/general-information) (that is MIT-opensourced) for WordPress, and where we bind all variables to view in the controller, we do not have support not for “Great_Widget extends WP_Widget” nor the Guttenberg blocks (we use shortcodes only instead, that does not have that ‘extend’ thing), besides that Guttenberg itself, is on React (MIT licensed), but that was troublesome for SolidMVC to fully support Guttenberg for commercial system via Envato and other portals, and even SolidMVC micro-framework port to Laravel framework.

**Comment by jeffpaul** [comment]
Noting here that this has been proposed on Make/Core including a process by which to gain approval for and implement dual-licensing for Gutenberg: https://make.wordpress.org/core/2020/09/21/proposal-dual-licensing-gutenberg-under-gpl-v2-0-and-mpl-v2-0/.

**Comment by techboyg5** [comment]
Isn't the MPL compatible with the GPL?

If it is, can't you choose to just license under the MPL?

**Comment by hsource** [comment]
@maxme - is there anything we can do to help this? I know it's probably not high-priority, but I'd love to contribute to the react-native-aztec package at work if it gets merged! For example, I realized that the support for toggling formatting is pretty incomplete, and I can see some quick fixes that I'd be happy to send PR's for

**Comment by maxme** [comment]
@hsource, the process to change the licensing scheme will take some time (months). 

In the meantime we can extract the react-native-aztec wrapper and switch it to MPL. Let me double check something first, I'll get back to you. We might do this quickly.

**Comment by maxme** [comment]
@hsource, we're going to revive this repo https://github.com/wordpress-mobile/react-native-aztec .We'll switch the license to MPL soon and pull the modifications from `WordPress/gutenberg` in there.

**Comment by hsource** [comment]
@maxme - that sounds great! Thanks for the update. I'm looking forward to contributing once it's revived.

**Comment by dngconsulting** [comment]
@maxme Is there any news on this topic ? The licence has changed but how to integrate react-native-aztec in a project without pulling all Gutenberg things ? Is there any dedicated npm package ? thanks.

**Comment by mchowning** [comment]
👋 @dngconsulting 

> Is there any news on this topic ? The licence has changed but how to integrate react-native-aztec in a project without pulling all Gutenberg things ? Is there any dedicated npm package ?

The license has been changed on the  `react-native-aztec` repository, and the code in that repository does not depend on Gutenberg, so you shouldn't need to pull anything from Gutenberg to use it. 

With that said, we have not ["pull[ed] the modifications from `WordPress/gutenberg`"](https://github.com/WordPress/gutenberg/issues/23651#issuecomment-723443955) to the `react-native-aztec` repository yet, so the code there is out-of-date. I'll work on getting that done, but it will probably be a couple of weeks.

**Comment by dngconsulting** [comment]
@mchowning Good to hear 👏 . But react-native-aztec is not published on npm registry, how do you 'npm install" it ?

**Comment by mchowning** [comment]
> react-native-aztec is not published on npm registry, how do you 'npm install" it ?

I'm not planning to publish the `react-native-aztec` repo to npm since it is just a temporary measure until Gutenberg is dual-licensed. Once the license is updated I'd be happy to publish the package from the Gutenberg repository if that would be helpful.

**Comment by pxpeterxu** [comment]
@dngconsulting I just revisited this thread after a long while, but if you want to install it, it's possible to install directly from a GitHub repository: try:

```sh
npm install wordpress-mobile/react-native-aztec#trunk
// Or with yarn
yarn add wordpress-mobile/react-native-aztec#trunk
```

**Comment by mokagio** [comment]
The license has been updated to `GPL-2.0-or-later`:

https://github.com/WordPress/gutenberg/blob/912dfe01d9745d92018b9a08f52c1551b07df1e3/package.json#L7

Should this issue be closed?

**Comment by swissspidy** [comment]
@mokagio no, as that is not related

**Comment by mokagio** [comment]
Thanks for the clarification @swissspidy
